server/utils/appInsightsUtils.ts

```diff
@@ -1,13 +1,43 @@
 import type { TelemetryClient } from 'applicationinsights'
-import { DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import { Contracts, DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import type { EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
 
 import { buildNumber, packageData } from '../applicationVersion'
+import type { User } from '../data'
+
+type ContextObjectWithUser = {
+  res?: {
+    locals?: {
+      user: Partial<User>
+    }
+  }
+}
+
+export type ContextObject = Record<string, ContextObjectWithUser> | undefined
 
 export default class AppInsightsUtils {
+  static addUserDataToRequests = (envelope: EnvelopeTelemetry, contextObjects: ContextObject): boolean => {
+    const isRequest = envelope.data.baseType === Contracts.TelemetryTypeString.Request
+    if (isRequest) {
+      const { username, activeCaseLoadId } = contextObjects?.['http.ServerRequest']?.res?.locals?.user || {}
+      if (username && envelope.data.baseData) {
+        const { properties } = envelope.data.baseData
+        // eslint-disable-next-line no-param-reassign
+        envelope.data.baseData.properties = {
+          activeCaseLoadId,
+          username,
+          ...properties,
+        }
+      }
+    }
+    return true
+  }
+
   static buildClient = (name = this.defaultName()): TelemetryClient | null => {
     if (process.env.APPLICATIONINSIGHTS_CONNECTION_STRING) {
       defaultClient.context.tags['ai.cloud.role'] = name
       defaultClient.context.tags['ai.application.ver'] = this.version()
+      defaultClient.addTelemetryProcessor(this.addUserDataToRequests)
```

Comment:
Does `this.` work here? I tried that before in another newly classed file and ended up using `ClassName.` instead since `this.`, which wasn't working as I'd expected from one class method to another